Threat Intelligence: Update name and descriptions of definitions, parameters

hunngu-ms opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ X I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

hunngu-ms committed:
Update API descriptions

AutorestCI commented:
/azp run automation - sdk

RazPopov commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -3627,8 +3627,8 @@
-        "description": "Create a threat intelligence.",
-        "operationId": "CreateThreatIntelligence",
+        "description": "Create a new threat intelligence indicator.",
+        "operationId": "ThreatIntelligenceIndicator_CreateIndicator",
```
Do we need to say Indicator twice?
Why not ThreatIntelligenceIndicator_Create ?

hunngu-ms replied:
ThreatIntelligenceIndicator_CreateIndicator:
In the first part, Indicator is required because TI Connector already took ThreatIntelligence, automatic checks does not allow.
In the second part, CreateIndicator is required this is createIndicator api, validation will give us warning if we dont have that

RazPopov replied:
Ah got it. So we're forced to these names,

inegroiu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -3674,14 +3674,14 @@
           "ThreatIntelligence"
         ],
-        "description": "Gets all threat intelligence.",
+        "description": "View all threat intelligence indicators.",
```
"List (or Get) all threat intelligence indicators" instead of View?

inegroiu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -3716,11 +3716,11 @@
           },
           "default": {
-            "description": "Error response describing why the operation failed.",
+            "description": "Error response describing why the operation failed to view indicators.",
```
faile to "list or get" indicators?

inegroiu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -3780,15 +3780,15 @@
-        "description": "Upsert a threat intelligence.",
-        "operationId": "ThreatIntelligenceIndicatorUpsert_Create",
+        "description": "Update a threat Intelligence indicator.",
+        "operationId": "ThreatIntelligenceIndicator_Create",
```
ThreatIntelligenceIndicator_Create? Not Update, or UpdateIndicator?

RazPopov replied:
This is strange. We have an operation called ThreatIntelligenceIndicator_CreateIndicator and another one ThreatIntelligenceIndicator_Create. I believe this should be resolved somehow... is it not?

hunngu-ms replied:
"Put" requires _Create

inegroiu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -4056,21 +4053,21 @@
           },
           {
-            "$ref": "#/parameters/ThreatIntelligenceReplaceTagsModel"
+            "$ref": "#/parameters/ThreatIntelligenceReplaceTags"
```
Not ThreatIntelligenceIndicator_ReplaceTags?

hunngu-ms replied:
This is parameter, not OperationID

hunngu-ms committed:
Resolve Comments

AutorestCI commented:
/azp run automation - sdk

ArcturusZhang commented:
Hi @hunngu-ms this PR introduces breaking change to an api-version which has been in preview for over 1 year. Per [Retirement of Preview](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews) guidance, products in Preview for over one year need to move to GA or retire; require CVP exemption after 1 year in preview.

Also for the breaking changes, please follow the steps in [this documentation](https://aka.ms/AzBreakingChangesPolicy) to request a review. Thanks.

ArcturusZhang commented:
Hi @hunngu-ms any updates?

hunngu-ms commented:
> Hi @hunngu-ms any updates?

@ArcturusZhang : The APIs that we updated are new APIs we added 2 weeks ago for public preview. There is almost nobody using them yet.

hunngu-ms committed:
Update parameter name of create/update indicator apis

AutorestCI commented:
/azp run automation - sdk

ArcturusZhang commented:
Hi @hunngu-ms any updates? Also could we please fix the model validation failure?

hunngu-ms committed:
Fix example issue

ArcturusZhang commented:
Altered a bit with the breaking change approved label to see if we could workaround the situation that breaking change CI is always in progress...
But we could not......

ArcturusZhang commented:
/azp run

jianyexi commented:
/azp run
